qcfractal/procedures/base.py

```diff
@@ -73,12 +73,16 @@ def retrieve_outputs(self, rdata):
             self.logger.warning(f"Found uncompressed error for result id {rdata['id']}")
             error = KVStore(data=rdata["error"])
 
-        # Now add to the database and set the ids in the diction
-        outputs = [stdout, stderr, error]
-        stdout_id, stderr_id, error_id = self.storage.add_kvstore(outputs)["data"]
-        rdata["stdout"] = stdout_id
-        rdata["stderr"] = stderr_id
-        rdata["error"] = error_id
+        # Now add to the database and set the ids in the dictionary
```

Comment:
Originally performed this way for bulk insert performance, I'm unsure if you will still hit these bottlenecks.